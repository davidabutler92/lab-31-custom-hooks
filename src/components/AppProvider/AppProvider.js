import React, { createContext, useState } from 'react';
import styles from '../../theme';

export const AppContext = createContext();

const AppProvider = ({ children }) => {
  const [theme, setTheme] = useState('light');

  const mode = styles[theme];
  const value = { setTheme, theme };

  return (
    <AppContext.Provider value={value}>
      <div style={mode}>{children}</div>
    </AppContext.Provider>
  );
};

export default AppProvider;
