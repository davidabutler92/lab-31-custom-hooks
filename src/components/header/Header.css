.header {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100vw;
  height: 10vh;
  background-color: lightgrey;
  border-radius: 10px;
  box-shadow: darkgray 10px 2px 20px 10px;
}

.link {
  text-decoration: none;
  font-size: 2em;
}
