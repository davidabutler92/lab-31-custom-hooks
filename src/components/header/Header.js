import React from 'react';
import { Link } from 'react-router-dom';
import styles from './Header.css';

export default function Header() {
  return (
    <div className={styles.header}>
      <Link className={styles.link} to="/">
        Home Page
      </Link>
    </div>
  );
}
