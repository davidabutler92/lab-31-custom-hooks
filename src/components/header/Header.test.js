import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Header from './Header';
import { MemoryRouter } from 'react-router-dom';
import AppProvider from '../AppProvider/AppProvider';

describe('Header component', () => {
  afterEach(() => cleanup());
  it('renders the header', () => {
    const { asFragment } = render(
      <AppProvider>
        <MemoryRouter>
          <Header />
        </MemoryRouter>
      </AppProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
