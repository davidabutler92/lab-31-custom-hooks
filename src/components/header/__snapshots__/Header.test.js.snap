// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders the header 1`] = `
<DocumentFragment>
  <div
    style="background-color: white; color: black;"
  >
    <div
      class="header"
    >
      <a
        class="link"
        href="/"
      >
        Home Page
      </a>
      <button>
        Click
      </button>
    </div>
  </div>
</DocumentFragment>
`;
